layers/core_checks/cc_device_memory.cpp

```diff
@@ -2678,3 +2726,97 @@ bool CoreChecks::PreCallValidateBindTensorMemoryARM(VkDevice device, uint32_t bi
+    const auto bpr_match = std::find_if(bp_requirements.begin(), bp_requirements.end(), [bind_info](const VkDataGraphPipelineSessionBindPointRequirementARM& bpr) {
+        return bpr.bindPoint == bind_info.bindPoint;
+    });
+    if (bpr_match == bp_requirements.end()) {
+        skip |= LogError("VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-bindPoint-09786", objlist, loc.dot(Field::bindPoint),
+            "bindPoint (%s) not found in requirements.",
+            string_VkDataGraphPipelineSessionBindPointARM(bind_info.bindPoint));
+    } else {
+        if (bind_info.objectIndex > bpr_match->numObjects) {
+            skip |= LogError(
+                "VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-objectIndex-09805", objlist, loc.dot(Field::objectIndex),
+                "objectIndex (%" PRIu32 ") is greater than numObjects (%" PRIu32 ") defined for bindPoint (%s)",
+                bind_info.objectIndex, bpr_match->numObjects,
+                string_VkDataGraphPipelineSessionBindPointARM(bind_info.bindPoint));
+        }
+    }
+
+    /* no point continuing if the bindpoint isn't valid: either not found or the index will cause a buffer overrun later on */
+    if (skip) {
+        return true;
+    }
+
+    const auto& mem_reqs_map = session_state->MemReqsMap();
+    const auto& bound_memory_map = session_state->BoundMemoryMap();
+    auto mem_info = Get<vvl::DeviceMemory>(bind_info.memory);
+    ASSERT_AND_RETURN_SKIP(mem_info);
+    if (bound_memory_map.find(bind_info.bindPoint) != bound_memory_map.end()) {
+        for (const auto &bound_mem : bound_memory_map.at(bind_info.bindPoint)) {
+            if (bound_mem.memory_state->VkHandle() == bind_info.memory) {
+                skip |= LogError("VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-session-09785",
+                    objlist, loc.dot(Field::bindPoint), "attempting to bind %s to %s which has already been bound to %s.",
+                    FormatHandle(mem_info->Handle()).c_str(),
+                    FormatHandle(session_state->Handle()).c_str(),
+                    FormatHandle(bound_mem.memory_state->Handle()).c_str());
+            }
+        }
+    }
+
+    skip |= ValidateInsertMemoryRange(VulkanTypedHandle(bind_info.session, kVulkanObjectTypeDataGraphPipelineSessionARM),
+                                        *mem_info, bind_info.memoryOffset, loc.dot(Field::memoryOffset));
+    if (mem_reqs_map.find(bind_info.bindPoint) != mem_reqs_map.end()) {
+        const auto &mem_reqs = mem_reqs_map.at(bind_info.bindPoint)[bind_info.objectIndex];
+        skip |= ValidateMemoryTypes(*mem_info, mem_reqs.memoryTypeBits, loc,
+                                    "VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-memory-09788");
+        if (SafeModulo(bind_info.memoryOffset, mem_reqs.alignment) != 0) {
+            skip |= LogError("VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-memoryOffset-09789", objlist, loc,
```

Comment:
```suggestion
            skip |= LogError("VUID-VkBindDataGraphPipelineSessionMemoryInfoARM-memoryOffset-09789", objlist, loc.dot(Field::memoryOffset), "(%" PRIu64 ") must be an..."
```

also print out `mem_reqs.alignment` here